native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -353,21 +353,35 @@ private static BArray newJsonList() {
      * @param element XML element to extract attributes and namespaces
      */
     private static LinkedHashMap<String, String> collectAttributesAndNamespaces(BXmlItem element,
-                                                                                String namespaceDelimiter) {
+                                                                                String namespaceDelimiter,
+                                                                                boolean preserveNamespaces) {
         LinkedHashMap<String, String> attributeMap = new LinkedHashMap<>();
         BMap<BString, BString> attributesMap = element.getAttributesMap();
         Map<String, String> nsPrefixMap = getNamespacePrefixes(attributesMap);
 
         for (Map.Entry<BString, BString> entry : attributesMap.entrySet()) {
-            if (isNamespacePrefixEntry(entry)) {
-                addNamespacePrefixAttribute(attributeMap, entry, namespaceDelimiter);
+            if (preserveNamespaces) {
+                if (isNamespacePrefixEntry(entry)) {
+                    addNamespacePrefixAttribute(attributeMap, entry, namespaceDelimiter);
+                } else {
+                    addAttributePreservingNamespace(attributeMap, nsPrefixMap, entry, namespaceDelimiter);
+                }
             } else {
-                addAttributePreservingNamespace(attributeMap, nsPrefixMap, entry, namespaceDelimiter);
+                preserveNonNamespacesAttributes(attributeMap, nsPrefixMap, entry, namespaceDelimiter);
             }
         }
         return attributeMap;
     }
 
+    private static void preserveNonNamespacesAttributes(LinkedHashMap<String, String> attributeMap,
```

Comment:
Shall we split the map entry param `entry` into 2 params; `key` and `value` and do the same to `addAttributePreservingNamespace` function?

Reply:
Fixed in [b219501](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/405/commits/b219501819a353771f75084b7bfd559ca148d0a2)